tracks/javascript/exercises/leap/mentoring.md

```diff
@@ -1,27 +1,67 @@
+The point of this exercise is to teach the student to use:
+- [Remainder (%)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Arithmetic_Operators#Remainder_()), also known as modulo
+- [Logical operators](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_Operators)
+- [Operator precedence](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence)
+
 ### Reasonable solutions
 
 ```javascript
-export const isLeap = (year) => year % 4 === 0 && year % 100 !== 0 || year % 400 === 0;
+export const isLeap = (year) => year % 4 === 0 && (year % 100 !== 0 || year % 400 === 0);
 ```
+Variations check for truthy `year % m` and falsy `(!year % m)`.
 
 ```javascript
-export const isLeap = (year) => !(year % 4) && year % 100 || !(year % 400);
-```
+  function isDivisibleByFn(mod) {
+    return number => number % mod === 0;
+  }
+
+  const by4 = isDivisibleByFn(4);
+  const by100 = isDivisibleByFn(100);
+  const by400 = isDivisibleByFn(400);
+
+  export const isLeap = year => by4(year) && (!by10(year) || by400(year))
```

Comment:
Should students walk away from the Leap exercise with an understanding of currying or partial application? 

Consider the following solution as it seems more applicable to the majority of students at this stage in the track,

```JavaScript
function isDivisibleBy (year, mod) {
  return year % mod === 0;
}

export const isLeap = (year) => {
  const by4 = isDivisibleBy(year, 4);
  const by100 = isDivisibleBy(year, 100);
  const by400 = isDivisibleBy(year, 400);

  return by4 && (!by100 || by400);
}
```

Reply:
I simply copied what about 1/10 students had. I agree that your solution is easier to read, but it is not identical to the non-functional solution. You make all comparisons and the whole "order of operations and evaluations" is sorta pointless. Instead I combined what was there with what you wrote!